tests/lib/enableRendererCoverage.js

```diff
@@ -0,0 +1,273 @@
+/**
+ * Coverage collecting for all BrowserWindows
+ *
+ * Enable coverage collecting mechanism for the BrowserWindows which is generated
+ * throughout the gpii-app Integration tests.
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    ipcMain = require("electron").ipcMain,
+    BrowserWindow = require("electron").BrowserWindow;
+
+require("gpii-testem"); // needed for the coverage server `gpii.testem.coverage.express`
+
+/*
+ * - Run coverage server (gpii.testem.coverage.express) that will be collecting the renderer's coverage data
+ * - Ensure all BrowserWindows are loading instrumented code to introduce coverage collecting in the renderer processes' code
+ * - Ensure dialogs (BrowserWindows) are not destroyed before their coverage is collected
+ *     - currently the only dialogs that are being destroyed are the `closable` dialogs and we simply prevent
+ *     their destruction for until their coverage is collected which is done at the end of a test sequence;
+ *     - this is done by disabling both their mechanism for getting destroyed by being closed and getting destroyed with
+ *     their wrapping component destruction;
+ *     - Note that we don't destroy every dialog in the final step but only the ones that are `closable` and have their
+ *     wrapping component destroyed. This is so as these components might be used in some post test sequence code execution;
+ * - Collect coverage data from BrowserWindows after every test sequence and destroy postponed dialogs
+ *     - this is needed as we need to collect coverage data before the BrowserWindows are destructed;
+ *     - once a test sequence has finished snippet in all running BrowserWindows is executed - this sends coverage data
+ *       to the running coverage server which stores this BrowserWindow coverage inforamtion;
+ *     - Note that single BrowserWindows type (e.g. QSS) would get multiple coverage data
```

Comment:
maybe expand on what you mean by "multiple coverage data", i.e. one per test run, et cetera?

Reply:
nm, saw the next lines, seems fine.